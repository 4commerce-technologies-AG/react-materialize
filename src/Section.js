import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Section = ({ className, node, id, children, ...other }) => {
  let C = node;
  let classes = {
    section: true
  };

  return (
    <C {...other} id={id} className={cx(classes, className)}>
      {children}
    </C>
  );
};

Section.propTypes = {
  children: PropTypes.node,
  node: PropTypes.node,
  id: PropTypes.string,
  className: PropTypes.string
};

Section.defaultProps = {
  node: 'div'
};

export default Section;
