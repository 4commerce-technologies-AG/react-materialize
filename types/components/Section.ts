import * as React from 'react';

interface SectionProps {
  node?: React.ReactElement | React.ReactNode;
  id?: string;
  className?: string;
}

/**
 * React Materialize: Section
 */
declare const Section: React.FC<SectionProps>

export default Section;
