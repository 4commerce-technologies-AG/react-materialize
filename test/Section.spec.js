import React from 'react';
import { mount } from 'enzyme';
import Section from '../src/Section';

describe('<Section />', () => {
  let wrapper;

  test('renders', () => {
    wrapper = mount(<Section />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders children', () => {
    wrapper = mount(
      <Section>
        <p>nope</p>
      </Section>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
