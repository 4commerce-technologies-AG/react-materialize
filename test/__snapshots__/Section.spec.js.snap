// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> renders 1`] = `
<Section
  node="div"
>
  <div
    className="section"
  />
</Section>
`;

exports[`<Section /> renders children 1`] = `
<Section
  node="div"
>
  <div
    className="section"
  >
    <p>
      nope
    </p>
  </div>
</Section>
`;
